src/query/allowed_objects.c

```diff
@@ -14,18 +14,49 @@ static const char *const g_allowed_casts[] = {
     "ftod", "dtof",
     "int4_numeric", "float4_numeric", "float8_numeric",
     "numeric_float4", "numeric_float8",
+    "date_timestamptz",
     /**/
 };
 
-static const char *const g_allowed_builtins[] = {
+typedef struct FunctionWithName
+{
+  const char *name;
+  int primary_arg;
+} FunctionWithName;
+
+typedef struct FunctionWithOid
+{
+  Oid funcid;
+  int primary_arg;
+} FunctionWithOid;
+
+static const FunctionWithName g_allowed_builtins[] = {
     /* rounding casts */
-    "ftoi2", "ftoi4", "ftoi8", "dtoi2", "dtoi4", "dtoi8", "numeric_int4",
+    (FunctionWithName){.name = "ftoi2", .primary_arg = 0},
```

Comment:
Does dropping `.primary_arg = 0` cause it be zero anyway?

Reply:
it does indeed, but it's not a meaningful default in here, so I'm reluctant to leverage this

Reply:
Yes, `primary_arg` should always be explicit. The zero trick will work well for `.name` or `.oid` if they are mutually exclusive.